src/joystick.cc

```diff
@@ -0,0 +1,130 @@
+  Nan::SetMethod(target, "joystickIsButtonPressed", IsButtonPressed);
+  Nan::SetMethod(target, "joystickGetAxisPosition", GetAxisPosition);
+  Nan::SetMethod(target, "joystickGetIdentification", GetIdentification);
+  Nan::SetMethod(target, "joystickUpdate", Update);
+
+  Local<Object> axis = Nan::New<Object>();
+  for (auto it = gen::joystick_axis_atoi.begin();
+       it != gen::joystick_axis_atoi.end();
+       it++) {
+    Nan::Set(axis,
+             Nan::New<String>(it->first).ToLocalChecked(),
+             Nan::New(it->second));
+  }
+
+  Nan::Set(target, Nan::New<String>("joystickAxis").ToLocalChecked(), axis);
+}
+
+NAN_METHOD(IsConnected) {
+  if (info.Length() != 1) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  info.GetReturnValue().Set(sf::Joystick::isConnected(joystick_id));
+}
+
+NAN_METHOD(GetButtonCount) {
+  if (info.Length() != 1) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  info.GetReturnValue().Set(sf::Joystick::getButtonCount(joystick_id));
+}
+
+NAN_METHOD(HasAxis) {
+  if (info.Length() != 2) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  sf::Joystick::Axis axis = static_cast<sf::Joystick::Axis>(
+      Nan::To<sf::Int32>(info[1].As<v8::Int32>()).FromJust());
+  info.GetReturnValue().Set(sf::Joystick::hasAxis(joystick_id, axis));
+}
+
+NAN_METHOD(IsButtonPressed) {
+  if (info.Length() != 2) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  sf::Uint32 button = Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  info.GetReturnValue().Set(sf::Joystick::isButtonPressed(joystick_id, button));
+}
+
+NAN_METHOD(GetAxisPosition) {
+  if (info.Length() != 2) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  sf::Joystick::Axis axis = static_cast<sf::Joystick::Axis>(
+      Nan::To<sf::Int32>(info[1].As<v8::Int32>()).FromJust());
+  info.GetReturnValue().Set(sf::Joystick::getAxisPosition(joystick_id, axis));
+}
+
+NAN_METHOD(GetIdentification) {
+  if (info.Length() != 1) {
+    Nan::ThrowError("Invalid arguments count.");
+    return;
+  }
+
+  sf::Uint32 joystick_id =
+      Nan::To<sf::Uint32>(info[0].As<v8::Uint32>()).FromJust();
+  sf::Joystick::Identification identification =
+      sf::Joystick::getIdentification(joystick_id);
+
+  Local<Object> ret = Nan::New<Object>();
+  Nan::Set(ret,
+           Nan::New<String>("name").ToLocalChecked(),
+           Nan::New<String>(identification.name).ToLocalChecked());
```

Comment:
> node: symbol lookup error: /build/Release/binding.node: undefined symbol: _ZNK2sf6StringcvNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEv

We can't create `v8::String` from `sf::String` directly.

Reply:
Good to know, on my side it just built without having any issue, I'll be careful with that. (Might be some weird vc++ trickery)